### PR TITLE
Fix balance cache on switching to testnet

### PR DESCRIPTION
**Pull request by rasom**
Balance was considered as existing if there were cached balances for any networks. In result on switching to testnet blenches were not requested.  

needed for https://github.com/status-im/status-desktop/issues/13964

### CODE REVIEW

File: services/wallet/reader.go [hunk 1]
```diff
@@ -330,10 +329,33 @@ func (r *Reader) getWalletTokenBalances(ctx context.Context, addresses []common.
...
 	if !updateBalances {
 		for _, address := range addresses {
-			if _, ok := cachedTokens[address]; !ok {
+			if res, ok := cachedTokens[address]; !ok || len(res) == 0 {
 				updateAnyway = true
 				break
 			}
+
+			networkFound := map[uint64]bool{}
+			for _, chain := range chainIDs {
```
Comment (IvanBelyakoff) [inline services/wallet/reader.go]: I think this is not needed

File: services/wallet/reader.go [hunk 2]
```diff
@@ -330,10 +329,33 @@ func (r *Reader) getWalletTokenBalances(ctx context.Context, addresses []common.
...
+				for _, chain := range chainIDs {
+					if _, ok := token.BalancesPerChain[chain]; ok {
+						networkFound[chain] = true
+					}
+				}
+			}
+
+			for _, chain := range chainIDs {
+				if !networkFound[chain] {
+					updateAnyway = true
```
Comment (IvanBelyakoff) [inline services/wallet/reader.go:349]: break?

**Review by IvanBelyakoff — COMMENTED** [review]
Good job solving the issue! IMO it would be nice to put it in some func like `isBalanceUpdateNeeded`

**Review by alaibe — APPROVED** [review]

File: services/wallet/reader.go (hunk 2 above)
Comment (rasom) [reply services/wallet/reader.go:349]: well, might be there, does it really matter?

File: services/wallet/reader.go (hunk 1 above)
Comment (rasom) [reply services/wallet/reader.go]: you mean if I rewrite it without `ok` check?

Comment (IvanBelyakoff) [reply services/wallet/reader.go]: I mean I dont understand why you initialize all the map values to false. If you dont - networkFound[chain] will return false.

File: services/wallet/reader.go (hunk 2 above)
Comment (IvanBelyakoff) [reply services/wallet/reader.go:349]: normally if you don't need the loop to proceed, you put break

Comment (rasom) [reply services/wallet/reader.go:349]: fixed

File: services/wallet/reader.go (hunk 1 above)
Comment (rasom) [reply services/wallet/reader.go]: fixed

**Review by IvanBelyakoff — APPROVED** [review]

**Pull request merged**